### PR TITLE
Sphinx improvements

### PR DESCRIPTION
**Pull request by webknjaz**
Added various Sphinx improvements:
* extlinks (+demo usage)
* intersphinx
* theme options
* RTD yaml config
* docs deps moved to extras

#### PR Type

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -129,14 +151,19 @@
...
+    'github_user': github_repo_org,
+    'github_repo': github_repo_name,
     'github_button': True,
+    'github_banner': True,
+    'github_type': 'star',
+    'github_count': True,
     'badge_branch': 'master',
     'travis_button': True,
+    'codecov_button': True,
+    # 'analytics_id': ...,
```
Comment (gundalow) [inline doc/source/conf.py]: I have this set in the `Advanced Settings` in RTD project settings, not sure if you'd want it here as well
```suggestion
     'analytics_id': 'https://github.com/ansible/community/issues/391',
```

Comment (webknjaz) [reply doc/source/conf.py]: @gundalow it looks like you've copy-pasted the wrong thing :)

P.S. I usually prefer file-based configs as Git tracks them better throughout the history + everything is in one place.

Comment (gundalow) [reply doc/source/conf.py]: Urgh, need coffee

Comment (webknjaz) [reply doc/source/conf.py]: https://molecule.readthedocs.io/en/latest/_static/readthedocs-data.js has
```js
var READTHEDOCS_DATA = {
    [...]
    global_analytics_code: 'UA-17997319-1',
    user_analytics_code: 'UA-128382387-1'
};
```
I guess it's the last one? So it'd be:
```suggestion
    'analytics_id': 'UA-128382387-1',
```

Comment (gundalow) [reply doc/source/conf.py]: Yup, `UA-128382387-1` is correct.

**Comment by webknjaz** [comment]
@gundalow is this blocked by #1541?

**Comment by gundalow** [comment]
Merged and follow up fix https://github.com/ansible/molecule/pull/1551 resolves `Problem parsing YAML configuration. Invalid "requirements_file": path doc-requirements.txt does not exist`

Docs rebuild and 
* confirmed that `intersphinx_mapping` mappings such as https://pip.pypa.io/en/latest/usage/ work
* confirmed that `gh` links in CHANGELOG.rst work

**Pull request merged**